### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by subash-poudel**
Tasks:

1. Add `eslint-config-leapfrog` library
2. Separate routes for different controllers

### CODE REVIEW

File: src/controllers/users.js [hunk 1]
```diff
@@ -1,58 +1,77 @@
...
- * GET /api/users
+ * Get all users
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Function} next
  */
-router.get('/', (req, res, next) => {
-  userService
-    .getAllUsers()
-    .then(data => res.json({ data }))
```
Comment (mesaugat) [inline src/controllers/users.js]: I kept using `then` because it looks more cleaner to me.

Comment (mesaugat) [reply src/controllers/users.js]: Also, it doesn't need to be `async`.

File: src/controllers/users.js [hunk 2]
```diff
@@ -1,58 +1,77 @@
-import { Router } from 'express';
 import HttpStatus from 'http-status-codes';
 import * as userService from '../services/userService';
-import { findUser, userValidator } from '../validators/userValidator';
-
-const router = Router();
 
 /**
- * GET /api/users
+ * Get all users
```
Comment (mesaugat) [inline src/controllers/users.js]: Full stop at the end and a new line after that.

File: src/controllers/users.js [hunk 3]
```diff
@@ -1,58 +1,77 @@
...
+    const data = await userService.getUser(req.params.id);
+    res.json({ data });
+  } catch (error) {
+    next(error);
+  }
+}
 
 /**
- * POST /api/users
+ * Create a new user
```
Comment (mesaugat) [inline src/controllers/users.js]: New line and a full stop.

File: src/controllers/users.js [hunk 4]
```diff
@@ -1,58 +1,77 @@
...
+    const data = await userService.createUser(req.body);
+    res.status(HttpStatus.CREATED).json({ data });
+  } catch (error) {
+    next(error);
+  }
+}
 
 /**
- * PUT /api/users/:id
+ * Update a user
```
Comment (mesaugat) [inline src/controllers/users.js]: New line and a full stop.

File: src/controllers/users.js [hunk 5]
```diff
@@ -1,58 +1,77 @@
...
+    const data = await userService.updateUser(req.params.id, req.body);
+    res.json({ data });
+  } catch (error) {
+    next(error);
+  }
+}
 
 /**
- * DELETE /api/users/:id
+ * Delete a user
```
Comment (mesaugat) [inline src/controllers/users.js]: same.

File: src/routes/userRoutes.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * GET /api/users
+ */
+router.get('/', userController.fetchAll);
+
+/**
+ * GET /api/users/:id
+ */
+router.get('/:id', userController.fetchById);
```
Comment (mesaugat) [inline src/routes/userRoutes.js]: This is clean and nice: 💯

File: src/controllers/users.js (hunk 1 above)
Comment (subash-poudel) [reply src/controllers/users.js]: 🆗

File: src/controllers/users.js (hunk 2 above)
Comment (subash-poudel) [reply src/controllers/users.js]: 🆗

File: src/controllers/users.js (hunk 5 above)
Comment (subash-poudel) [reply src/controllers/users.js]: 🆗

File: src/controllers/users.js (hunk 4 above)
Comment (subash-poudel) [reply src/controllers/users.js]: 🆗

File: src/controllers/users.js (hunk 3 above)
Comment (subash-poudel) [reply src/controllers/users.js]: :ok

File: src/routes/userRoutes.js (hunk 6 above)
Comment (subash-poudel) [reply src/routes/userRoutes.js]: 🆗

**Comment by mesaugat** [comment]
Looks good. Thanks @subash-poudel 👍 

I'll fix the tests and merge it.

**Pull request merged**